hcaas/resource_hcaas_group.go

```diff
@@ -66,21 +70,35 @@ func resourceHcaasGroupCreate(ctx context.Context, d *schema.ResourceData, meta
 
 	req.Header.Set("Authorization", provider.Token)
 
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return diag.FromErr(err)
-	}
+	err = retry.RetryContext(ctx, d.Timeout(schema.TimeoutCreate)-time.Minute, func() *retry.RetryError {
+		resp, err := http.DefaultClient.Do(req)
 
-	defer resp.Body.Close()
+		if err != nil {
+			if strings.Contains(err.Error(), "event locked") {
```

Comment:
@tcarreira could you create a generic function like

```
func retryRequest(ctx context.Context,  req *http.Request) (http.Response, error) {
...
}

Reply:
@wpjunior did it. What do you think?